sdk/samples/iot/hub/src/sample_pnp_thermostat_component.h

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_SAMPLE_PNP_THERMOSTAT_COMPONENT_H
+#define _az_SAMPLE_PNP_THERMOSTAT_COMPONENT_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
Why do we have this?

Reply:
might just be muscle memory from before. seems that `_az_cfg_prefix.h` has it too but didn't think to add that

Reply:
I mean should we be adding `_az_cfg_prefix.h` to sample headers?

Reply:
No, I don't think we should add non-public headers in samples.

I think my question should be directed to folks working on core. @antkmsft, do you recall why we have this and do we still need it?

https://github.com/Azure/azure-sdk-for-c/blob/f50b90403b8b370b55c14cfc60f6ebe013a32464/sdk/inc/azure/core/_az_cfg_prefix.h#L15-L18

Looks like it was added in https://github.com/Azure/azure-sdk-for-c/pull/31/files#diff-9bea604bc146230b0c86a481fa559384R9-R11 and then moved to the config files: https://github.com/Azure/azure-sdk-for-c/pull/27/files#diff-ed9d3d869305cdb374fc7243b8c9c5c8R4

Reply:
`extern "C"` allows customers to include our headers from C++ compilers and build/link them.
This is C SDK, but I don't think we should drop having the ability to be interact with C++. Sometimes phave C++ compilers, but use it as "advanced C", or even just use the C89 subset of features.

Reply:
Thanks @antkmsft. Have we recently tried to build/link the SDK using a C++ compiler? We should check if that actually works if that is a use case we want to continue to support.